File: practice-algos/product_of_array_except_self/gpt_sol.py
def productExceptSelf(nums):
    n = len(nums)
    left, right, answer = [0]*n, [0]*n, [0]*n

    left[0] = 1
    for i in range(1, n):
        left[i] = left[i - 1] * nums[i - 1]

    right[n - 1] = 1
    for i in range(n - 2, -1, -1):
        right[i] = right[i + 1] * nums[i + 1]

    for i in range(n):
        answer[i] = left[i] * right[i]

    return answer

File: practice-algos/product_of_array_except_self/gpt_sol.js
function productExceptSelf(nums) {
    let n = nums.length;
    let left = new Array(n), right = new Array(n), answer = new Array(n);

    left[0] = 1;
    for (let i = 1; i < n; i++) {
        left[i] = left[i - 1] * nums[i - 1];
    }

    right[n - 1] = 1;
    for (let i = n - 2; i >= 0; i--) {
        right[i] = right[i + 1] * nums[i + 1];
    }

    for (let i = 0; i < n; i++) {
        answer[i] = left[i] * right[i];
    }

    return answer;
}
